lua/nvim-tree/actions/fs/remove-file.lua

```diff
@@ -22,17 +22,17 @@ end
 
 local function clear_buffer(absolute_path)
   local bufs = vim.fn.getbufinfo { bufloaded = 1, buflisted = 1 }
+  local tree_winnr = vim.api.nvim_get_current_win()
   for _, buf in pairs(bufs) do
     if buf.name == absolute_path then
       if buf.hidden == 0 and (#bufs > 1 or view.View.float.enable) then
-        local winnr = vim.api.nvim_get_current_win()
         vim.api.nvim_set_current_win(buf.windows[1])
         vim.cmd ":bn"
-        if not view.View.float.enable then
-          vim.api.nvim_set_current_win(winnr)
-        end
       end
       vim.api.nvim_buf_delete(buf.bufnr, { force = true })
+      if not view.View.float.quit_on_focus_loss then
```

Comment:
found that `vim.api.nvim_buf_delete` does not delete the buffer if there is no focus on it, so firstly delete buffer then return focus.